### PR TITLE
Sprint 33

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 33

## [3148 Remove link from order summary screen](https://github.com/flexion/ef-cms/issues/3148)
image
## [445 Prepare system for data migration](https://github.com/flexion/ef-cms/issues/445)
Screenshot N/A
## [1130 Docket Clerk: Edit Service Indicator](https://github.com/flexion/ef-cms/issues/1130)
image
## [3796 Court User: Add judges filter to deadline report](https://github.com/flexion/ef-cms/issues/3796)
image
## [3908 Calendaring: Any due date or pending item should put case on “blocked list”](https://github.com/flexion/ef-cms/issues/3908)
image
## [4307 Wireframes for service emails](https://github.com/flexion/ef-cms/issues/4307)
image
## [3910 Standing Pretrial Order/Notice](https://github.com/flexion/ef-cms/issues/3910)
image
## [3909 Make transcripts unavailable to parties for 90 days](https://github.com/flexion/ef-cms/issues/3909)
image
## [1912 Closing a Trial Session](https://github.com/flexion/ef-cms/issues/1912)
image
## [3958 Direct court user to signature page after saving a draft order](https://github.com/flexion/ef-cms/issues/3958)
image
## [436 Upload PDF documents to the system](https://github.com/flexion/ef-cms/issues/436)
image
## [3914 Docket Clerk: Ability to modify docket entry (QA)](https://github.com/flexion/ef-cms/issues/3914)\
image
## [3893 Create Court’s Testing and Production Environments (In Progress)](https://github.com/flexion/ef-cms/issues/3893)
Screenshot N/A
## [3896 E2E Round Table Testing (1/8)](https://github.com/flexion/ef-cms/issues/3896)
Screenshot N/A
## [4080 Docket Section Usability Testing (1/16)](https://github.com/flexion/ef-cms/issues/4080)
Screenshot N/A
## [4081 E2E Roundtable Testing (1/21)](https://github.com/flexion/ef-cms/issues/4081)
Screenshot N/A

### CODE REVIEW

**Comment by waldoj** [comment]
### PR Review Checklist

- [ ] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] run all tests and generate test data with `./test-all.sh`
- [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test each new story in all major browsers
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, looking for problems (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
		- [ ] review untested code rigorously
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I'll be using `README.md` as the place to attach PR-level comments.

File: shared/src/business/utilities/DateHandler.test.js [hunk 1]
```diff
@@ -28,6 +28,28 @@ describe('DateHandler', () => {
     });
   });
 
+  describe('calculateISODate', () => {
+    it('calculates dates with zero adjustment', () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/business/utilities/DateHandler.test.js]:
```
const calculateISODate = ({ dateString, howMuch = 0, units = 'days' }) => {
  if (!howMuch) return dateString;

  return prepareDateFromString(dateString, FORMATS.ISO)
    .add(howMuch, units)
    .toISOString();
};
```
When no additional params are used in `calculateISODate` it just returns the `dateString` as param as it was given. You are correct that this is not a date, but I think the intention was to make it clear that this was essentially a noop - returning _exactly_ what was given to it.

@waldoj  I can update the test to use a real date, leave it as-is and add a comment to this effect ☝️, or do nothing. Thoughts?

File: shared/src/persistence/s3/getDownloadPolicyUrl.test.js [hunk 2]
```diff
@@ -14,6 +14,21 @@ describe('getDownloadPolicyUrl', () => {
     expect(result).toEqual({ url: 'http://localhost' });
   });
 
+  it('makes a post request to the expected endpoint with the expected data', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/persistence/s3/getDownloadPolicyUrl.test.js]: Re-writing the test and its descriptions to make a lot more sense.

File: web-client/src/presenter/actions/EditDocketRecordEntry/setDocketEntryMetaFormForEditAction.test.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      },
+      state: {
+        caseDetail,
+      },
+    });
+
+    expect(result.state.form.servedPartiesCode).toEqual('');
+  });
+
+  it('gets computes the servedPartiesCode from documentDetail when NOT present on docketRecordEntry', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/actions/EditDocketRecordEntry/setDocketEntryMetaFormForEditAction.test.js]: copy/pasta, probably. Should just be "computes the servedPartiesCode..."

File: web-client/src/presenter/actions/FileDocument/computeFilingFormDateAction.test.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import { computeFilingFormDateAction } from './computeFilingFormDateAction';
+import { presenter } from '../../presenter';
+import { runAction } from 'cerebral/test';
+
+describe('computeFilingFormDateAction', () => {
+  beforeEach(() => {
+    presenter.providers.applicationContext = applicationContext;
+  });
+
+  it("computes the document's filingDate from date parts", async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/actions/FileDocument/computeFilingFormDateAction.test.js]: Our linter automagically changes:
`'this is a test\'s description'`
to
`"this is a test's description"`

*note the apostrophe.

File: web-client/src/presenter/actions/editUploadCourtIssuedDocument/setDocumentToFormAction.test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            documentToMatch,
+          ],
+        },
+        documentId: documentIdToEdit,
+      },
+    });
+    expect(result.state.form).toEqual(documentToMatch);
+  });
+
+  it('does nothing if documentId isdoes not match a document', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/actions/editUploadCourtIssuedDocument/setDocumentToFormAction.test.js]: The internet barely knows of this word "isdoes" - so we'll go with your suggestion, I guess. :P

File: web-client/src/presenter/computeds/formattedCaseDetail.test.js [hunk 6]
```diff
@@ -844,83 +848,174 @@ describe('formattedCaseDetail', () => {
...
+          isPetition: false,
+          isStatusServed: false,
+          showDocumentEditLink: true,
+          signedAtFormatted: undefined,
+          signedAtFormattedTZ: undefined,
         },
+      ]);
+    });
+
+    it('populates editLink for docket clerks', () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/computeds/formattedCaseDetail.test.js]: The test descriptions aren't entirely accurate. This editLink should not exist on draft documents. Updating to reflect that.

File: web-client/src/presenter/utilities/sequenceHelpers.test.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      const mockActionsArray = [jest.fn(), jest.fn(), jest.fn()];
+      const result = showProgressSequenceDecorator(mockActionsArray);
+
+      expect(result).toMatchObject([
+        expect.any(Function),
+        ...mockActionsArray,
+        expect.any(Function),
+      ]);
+    });
+    it('should take an array of actions (functions) and return a new array with same elements plus a function at beginning and end', () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/utilities/sequenceHelpers.test.js]: Yes. Updating.

File: shared/src/business/utilities/DateHandler.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply shared/src/business/utilities/DateHandler.test.js]: The new test description looks good.

File: web-client/src/presenter/actions/FileDocument/computeFilingFormDateAction.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply web-client/src/presenter/actions/FileDocument/computeFilingFormDateAction.test.js]: I hadn't noticed the apostrophe — my apologies.

**Comment by waldoj** [comment]
@JessicaMarine, I have reviewed this pull request and I recommend that it be accepted into the Court's staging environment.

**Pull request merged**